state-chain/pallets/cf-ingress-egress/src/lib.rs

```diff
@@ -3005,6 +3065,22 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			},
 		);
 	}
+
+	// TODO: Write test
+
+	// This should only be used if we're using ProcessedUpTo to track the block height.
+	pub fn active_deposit_channels_at(
+		opened_at_or_before: TargetChainBlockNumber<T, I>,
+		expires_after: TargetChainBlockNumber<T, I>,
+	) -> Vec<DepositChannelDetails<T, I>> {
+		debug_assert!(<T::TargetChain as Chain>::is_block_witness_root(opened_at_or_before));
+
+		DepositChannelLookup::<T, I>::iter_values()
+			.filter(|details| {
```

Comment:
This could get quite expensive - at any given time we might have a lot of open channels (for example right now on mainnet, 250), and it scales linearly with something that can be manipulated from outside (ie. by opening lots of channels). 

If we call this frequently I would suggest we consider caching this somehow (eg. storing a Vec of new/expired channels and `taking` them).

Reply:
As discussed - we can leave this for now.